source/extensions/common/workload_discovery/api.cc

```diff
@@ -31,9 +31,15 @@
 namespace Envoy::Extensions::Common::WorkloadDiscovery {
 
 namespace {
-Istio::Common::WorkloadMetadataObject convert(const istio::workload::Workload& workload) {
+Istio::Common::WorkloadMetadataObject convert(const istio::workload::Address& address) {
+  switch (address.type()) {
+    case istio::workload::Service:
+      return;
```

Comment:
This is incorrect, it still returns something. Maybe put ASSERT to ensure to ensure the type is right.